lib/matplotlib/contour.py

```diff
@@ -1213,10 +1211,17 @@ def _process_levels(self):
         # (Colorbar needs this even for line contours.)
         self._levels = list(self.levels)
 
+        tempLocator = Locator()
+        if self.logscale:
+            tempLocator = LogLocator()
+        if self.locator is not None:
+            tempLocator = self.locator
```

Comment:
I thought about this as well, but then it'll require too many different if-else statements in this methods to distinguish extend min, max, both and logscale is true or false. The caveat of if-else is that they spawn too many different code paths, and some of them are not even legal combinations. Using inheritance reduces these unnecessary code paths.